include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -189,25 +241,27 @@ void MjbotsControlLoop<log_type, input_type>::Run() {
 
     // If parallel mode, process the previous reply, then send command to keep cycle time up on pi3hat loop
     if(options_.parallelize_control_loop){
-      robot_->ProcessReply();
+      mjbots_interface_->ProcessReply();
       prev_msg_duration = message_duration_timer.tac();
       message_duration_timer.tic();
-      robot_->SendCommand();
+      mjbots_interface_->SendCommand();
     }
+
     // Calculate torques and log
-    CalcTorques();
+    CalcTorques();      //TODO should we give full control to the robot_ instead of feeding update thorugh?
```

Comment:
Lets rename CalcTorques() to Update() and have Update be up to the user what it does. They can call the robot update if they want or just do vanilla calc torques.

Reply:
Renamed in 2d28dd28bc0957d92233c7ff3f997166d212ee69.

Reply:
Because of bug in soft_start this will likely break code, but will be fixed before merged.

- [ ] Fix soft start timer - get rid of cycle count